ELSpaceTests/TestCases/ViewModels/ChooserActivityTypesViewModelSpec.swift

```diff
@@ -0,0 +1,108 @@
+                    let viewModel = sut.activityTypeViewModels[1] as! ActivityTypeViewModel
+                    expect(viewModel).to(equal(ActivityTypeViewModel(type: .vacation)))
+                }
+
+                it("should have third time report") {
+                    let viewModel = sut.activityTypeViewModels[2] as! ActivityTypeViewModel
+                    expect(viewModel).to(equal(ActivityTypeViewModel(type: .dayOff)))
+                }
+
+                it("should have fourth time report") {
+                    let viewModel = sut.activityTypeViewModels[3] as! ActivityTypeViewModel
+                    expect(viewModel).to(equal(ActivityTypeViewModel(type: .sickLeave)))
+                }
+
+                it("should have fifth time report") {
+                    let viewModel = sut.activityTypeViewModels[4] as! ActivityTypeViewModel
+                    expect(viewModel).to(equal(ActivityTypeViewModel(type: .conference)))
+                }
+            }
+
+            describe("switching types") {
+                var scheduler: TestScheduler!
+                var isTimeReportSelectedObserver: TestableObserver<Bool>!
+                var isVacationSelectedObserver: TestableObserver<Bool>!
+                var isDayOffSelectedObserver: TestableObserver<Bool>!
+                var isSickLeaveSelectedObserver: TestableObserver<Bool>!
+                var isConferenceSelectedObserver: TestableObserver<Bool>!
+
+                beforeEach {
+                    scheduler = TestScheduler(initialClock: 0)
+                    isTimeReportSelectedObserver = scheduler.createObserver(Bool.self)
+                    isVacationSelectedObserver = scheduler.createObserver(Bool.self)
+                    isDayOffSelectedObserver = scheduler.createObserver(Bool.self)
+                    isSickLeaveSelectedObserver = scheduler.createObserver(Bool.self)
+                    isConferenceSelectedObserver = scheduler.createObserver(Bool.self)
+                    _ = sut.activityTypeViewModels[0].isSelected.asObservable().subscribe(isTimeReportSelectedObserver)
+                    _ = sut.activityTypeViewModels[1].isSelected.asObservable().subscribe(isVacationSelectedObserver)
+                    _ = sut.activityTypeViewModels[2].isSelected.asObservable().subscribe(isDayOffSelectedObserver)
+                    _ = sut.activityTypeViewModels[3].isSelected.asObservable().subscribe(isSickLeaveSelectedObserver)
+                    _ = sut.activityTypeViewModels[4].isSelected.asObservable().subscribe(isConferenceSelectedObserver)
+                }
+
+                context("when selecting time report") {
+                    beforeEach {
+                        sut.activityTypeViewModels[0].isSelected.onNext(true)
+                    }
+
+                    it("should properly update state") {
+                        expect(isTimeReportSelectedObserver.events.last?.value.element).to(beTrue())
```

Comment:
It will be better to separate all expectations, because when something will fail on CI you will not know what goes wrong because all exceptations have this same description